### PR TITLE
Move CLI Into a Container

### PR DESCRIPTION
**Pull request by TylerJewell**
### What does this PR do?
1. Creates a codenvy/cli dockerfile
2. Moves the codenvy/version information into the CLI
3. Derives version information from the tag of the codenvy/cli image
4. Derives the values of CONFIG, INSTANCE and BACKUP folders from volume mounts
5. Uses docker/compose:1.8.1 docker image for compose instead of native tool
6. Fixes bugs that prevent using paths with spaces in the names.

### CODE REVIEW

**Comment by TylerJewell** [comment]
@benoitf @riuvshin - when I started this branch, it was originally off of the "docker" branch.  That branch has been merged.  I merged master with this branch, but it is picking up a huge history.  How can I suppress all of that history?

**Comment by benoitf** [comment]
@TylerJewell you have to merge master branch again and resolve conflicts.

File: dockerfiles/cli/codenvy.sh [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#   Tyler Jewell - Initial Implementation
+#
+
+init_host_ip() {
+  GLOBAL_HOST_IP=${GLOBAL_HOST_IP:=$(docker_run --net host codenvy/che-ip:nightly)}
```
Comment (TylerJewell) [inline dockerfiles/cli/codenvy.sh]: fixed

File: dockerfiles/cli/version/5.1.0/images [hunk 2]
```diff
@@ -7,6 +7,6 @@ IMAGE_NGINX=nginx:1.10-alpine
 IMAGE_SWARM=swarm:1.2.5
 IMAGE_REGISTRY=registry:2.5.0
 IMAGE_POSTGRES=postgres:9.5.4
-IMAGE_IP=eclipse/che-ip:nightly
```
Comment (TylerJewell) [inline dockerfiles/cli/version/5.1.0/images]: fixed

File: dockerfiles/cli/version/5.0.0-M3/images [hunk 3]
```diff
@@ -7,6 +7,6 @@ IMAGE_NGINX=nginx:1.10-alpine
 IMAGE_SWARM=swarm:1.2.5
 IMAGE_REGISTRY=registry:2.5.0
 IMAGE_POSTGRES=postgres:9.5.4
-IMAGE_IP=eclipse/che-ip:nightly
```
Comment (TylerJewell) [inline dockerfiles/cli/version/5.0.0-M3/images]: removed this utility from images as it's needed in bootstrap.

File: dockerfiles/cli/version/5.0.0-M2/images [hunk 4]
```diff
@@ -7,6 +7,6 @@ IMAGE_NGINX=nginx:1.10-alpine
 IMAGE_SWARM=swarm:1.2.5
 IMAGE_REGISTRY=registry:2.5.0
 IMAGE_POSTGRES=postgres:9.5.4
-IMAGE_IP=eclipse/che-ip:nightly
```
Comment (TylerJewell) [inline dockerfiles/cli/version/5.0.0-M2/images]: removed from images as it is a bootstrap image

**Pull request merged**